### PR TITLE
Workpackage template component

### LINKED ISSUE #2159: Work Package Templates - Hook up to create/edit WP pages
### Description

Blocked By #2148 #2156 

We want the work package templates to be accessible from the create/edit work packages pages. This ticket should be done after the get all work package template endpoint is done. See the mocks below for how the templates should be displayed. When a template is selected it should fill in all the corresponding fields. For blocked by it should search for a work package on the project associated with the CR that has the corresponding stage and name saved in the work package template's blocked by info (gonna be tricky)

### Acceptance Criteria

- [ ] When you click on a work package template it should fill in all the corresponding fields on the work package form
- [ ] If you're using it on an edit, it should fill around the existing values in the form (if an input already has a value in the input, it won't replace it)

### Proposed Solution

This one's gonna be a lot of react-hook-form trickiness, reference the docs for help, and ask a lot of questions!: https://react-hook-form.com/

### Mocks
image

### PR DESCRIPTION
**Pull request by AShyamkumar19**
## Changes

So far, created a new file called WorkPackageTemplateDetails where it's responsible for mapping out all the boxes in the WorkPackageTemplates. This react component is called in the WorkPackageView.

## Screenshots

image

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #2159

### CODE REVIEW

File: src/frontend/src/pages/WorkPackageForm/WorkPackageFormView.tsx [hunk 1]
```diff
@@ -119,6 +124,76 @@ const WorkPackageFormView: React.FC<WorkPackageFormViewProps> = ({
...
+  const {
+    data: workPackageTemplates,
+    isLoading: workPackageTemplateisLoading,
+    isError: workPackageTemplateisError,
+    error: workPackageTemplateError
+  } = useAllWorkPackageTemplates();
+  if (workPackageTemplateisLoading || !workPackageTemplates) return <LoadingIndicator />;
+  if (workPackageTemplateisError) return <ErrorPage message={workPackageTemplateError.message} />;
+
+  const testWorkPackageTemplate: WorkPackageTemplate[] = [
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/WorkPackageForm/WorkPackageFormView.tsx:136]: why is there test data in the actual form?

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
This looks mostly good, thanks for deleting the duplicate function, can you add screenshots to the PR and make sure to tag your commits

File: src/frontend/src/pages/WorkPackageForm/WorkPackageFormView.tsx (hunk 1 above)
Comment (AShyamkumar19) [reply src/frontend/src/pages/WorkPackageForm/WorkPackageFormView.tsx:136]: There was no seed data for WorkPackageTemplate, and there was no create WorkPackageTemplate endpoint so I couldn't create the seed data and wasn't sure if there was another way to create seed data. So for testing purposes, I just put the test data there. So, once that's created, I can create seed data.

Comment (Peyton-McKee) [reply src/frontend/src/pages/WorkPackageForm/WorkPackageFormView.tsx:136]: hrmm. I think @bderbs30 is working on the create and edit endpoints

Comment (bderbs30) [reply src/frontend/src/pages/WorkPackageForm/WorkPackageFormView.tsx:136]: I'm pretty sure I'm done I'll request review

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
Make Seed Data changes now

**Pull request closed without merging**